src/run.ts

```diff
@@ -95,11 +97,49 @@ function getDockerSecretArguments(secretName: string): string[] {
 
 function getGenericSecretArguments(secretName: string): string[] {
     const secretArguments = core.getInput('arguments');
+    const parsedArgument = fromLiteralsToFromFile(secretArguments);
     let args = ['create', 'secret', 'generic', secretName];
-    args.push(...argStringToArray(secretArguments));
+    args.push(...argStringToArray(parsedArgument));
     return args;
 }
 
+function fromLiteralsToFromFile(secretArguments: string): string {
+    const parsedArgument = secretArguments.split("--").reduce((argumentsBuilder, argument) => {
+        if (argument && !argument.startsWith("from-literal=")) argumentsBuilder += " --" + argument;
+        if (argument && argument.startsWith("from-literal=")) {
```

Comment:
Shouldn't this be else?

Reply:
Both works, this looked much cleaner. Let me add the else.